mrmustard/lab_dev/states/base.py

```diff
@@ -351,6 +333,23 @@ def fock_distribution(self, cutoff: int) -> ComplexTensor:
         else:
             return math.reshape(math.abs(math.diag_part(fock_array)), (-1,))
 
+    def normalize(self) -> State:
+        r"""
+        Returns a rescaled version of the state such that its probability is 1.
+        """
+        probability = self.probability
+        if probability.shape != () and isinstance(self.ansatz, PolyExpAnsatz):
+            delta = len(self.ansatz.c.shape) - len(probability.shape)
```

Comment:
I have added core_dims in one of the other PRs, so we will be able to simplify this too